lib/PuppeteerSharp/Frame.cs

```diff
@@ -406,31 +406,40 @@ public Task<string[]> SelectAsync(string selector, params string[] values)
         /// <seealso cref="Page.AddScriptTagAsync(string)"/>
         public async Task<ElementHandle> AddScriptTag(AddTagOptions options)
         {
-            const string addScriptUrl = @"async function addScriptUrl(url) {
+            const string addScriptUrl = @"async function addScriptUrl(url, type) {
               const script = document.createElement('script');
               script.src = url;
+              if(type)
+                script.type = type;
               document.head.appendChild(script);
               await new Promise((res, rej) => {
                 script.onload = res;
                 script.onerror = rej;
               });
               return script;
             }";
-            const string addScriptContent = @"function addScriptContent(content) {
+            const string addScriptContent = @"function addScriptContent(content, type = 'text/javascript') {
               const script = document.createElement('script');
-              script.type = 'text/javascript';
+              script.type = type;
               script.text = content;
               document.head.appendChild(script);
               return script;
             }";
 
+            async Task<ElementHandle> AddScriptTagPrivate(string script, string urlOrContent, string type)
```

Comment:
Let's add `private` here

Reply:
I don't like much the `Private` suffix here. This could be a private override right?

Reply:
It's a local method

Reply:
🤓